File: desktop/src/features/onboarding/forkGates.ts
/**
 * Fork configuration gates for the onboarding product layer.
 *
 * The Lunar Park fork ships without Buzz's bundled welcome experience — the
 * private Welcome channel, the welcome team/canvas seeding, and the kickoff
 * conversation it drives. The upstream code paths stay intact so rebases stay
 * cheap; this gate only controls whether they run.
 *
 * Set `VITE_BUZZ_WELCOME_EXPERIENCE=1` (or `true`) at build time to restore
 * the upstream behavior.
 */
export function welcomeExperienceEnabled(
  value: string | undefined = import.meta.env?.VITE_BUZZ_WELCOME_EXPERIENCE as
    | string
    | undefined,
): boolean {
  const normalized = value?.trim();
  return normalized === "1" || normalized === "true";
}

File: desktop/src-tauri/src/fork_gates.rs
//! Fork configuration gates.
//!
//! The Lunar Park fork disables some upstream boot-time behaviors by
//! default; each gate here re-enables one via an explicit env opt-in. The
//! upstream code paths stay intact — gates only control whether they run.

/// Whether boot-time voice (STT/TTS) model downloads are enabled.
///
/// Fork default is disabled; only an explicit `1` or `true` (from
/// `BUZZ_VOICE_MODEL_AUTODOWNLOAD`) restores the upstream fetch-at-boot
/// behavior. Models still download on-demand when transcription is used.
pub fn voice_model_autodownload_enabled(value: Option<&str>) -> bool {
    matches!(value.map(str::trim), Some("1") | Some("true"))
}

#[cfg(test)]
mod voice_model_gate_tests {
    use super::voice_model_autodownload_enabled;

    #[test]
    fn autodownload_is_disabled_by_default() {
        assert!(!voice_model_autodownload_enabled(None));
        assert!(!voice_model_autodownload_enabled(Some("")));
        assert!(!voice_model_autodownload_enabled(Some("0")));
        assert!(!voice_model_autodownload_enabled(Some("false")));
    }

    #[test]
    fn autodownload_enables_on_explicit_opt_in() {
        assert!(voice_model_autodownload_enabled(Some("1")));
        assert!(voice_model_autodownload_enabled(Some("true")));
        assert!(voice_model_autodownload_enabled(Some(" 1 ")));
    }
}
